cloud/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.9-slim AS base
+
+LABEL "repository"="https://github.com/nitrictech/nitric"
+LABEL "homepage"="https://nitric.io"
+LABEL "maintainer"="Nitric Team <team@nitric.io>"
+LABEL org.opencontainers.image.description="Nitric common deployer, in a Docker container."
+
+# Install deps all in one step
+RUN apt-get update -y && \
+  apt-get install -y \
+  apt-transport-https \
+  build-essential \
+  ca-certificates \
+  curl \
+  git \
+  gnupg \
+  software-properties-common \
+  wget \
+  unzip && \
+  # Get all of the signatures we need all at once.
+  curl -fsSL https://download.docker.com/linux/debian/gpg          | apt-key add - && \
+  # Add additional apt repos all at once
+  echo "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"      | tee /etc/apt/sources.list.d/docker.list           && \
+  # Install second wave of dependencies
+  apt-get update -y && \
+  apt-get install -y \
+  docker-ce && \
```

Comment:
Thinking of slimming things down a bit. Do we actually need full docker-ce? Or could we get away with just docker-cli with the socket forwarding we'll need to do anyway?